### ISSUE TITLE
LIBRENMS_DISTRIBUTED_POLLER_GROUP not changing poller group

### ISSUE THREAD

**Issue by BraydenBrekke** [librenms/docker#91]
### Behaviour

Distributed Poller Group ID is being set to default, Even when LIBRENMS_DISTRIBUTED_POLLER_GROUP is set. 

#### Steps to reproduce this issue

1. Set environment variable within docker compose to LIBRENMS_DISTRIBUTED_POLLER_GROUP=1 in the librenms container

#### Expected behaviour

> The Distributed Poller Group ID should not be defaulting to 0, but instead to the Poller Group ID defined in the variable above

#### Actual behaviour

> Poller Group ID is being set to default

### Configuration

* Docker version (type `docker --version`) : 19.03.12
* Docker compose version if applicable (type `docker-compose --version`) : 1.26.0
* Platform (Debian 9, Ubuntu 18.04, ...) : Ubuntu 18.04
* System info (type `uname -a`) : 
Linux itssdocker 4.15.0-108-generic #109-Ubuntu SMP Fri Jun 19 11:33:10 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

* Include all necessary configuration files : `docker-compose.yml`, `.env`, ...
### Docker-Compose.yml
```
version: "3.5"
services:
  memcached:
    image: memcached:alpine
    container_name: librenms_memcached
    ports:
      - target: 11211
        published: 11211
        protocol: tcp
    environment:
      - "TZ=${TZ}"
    restart: always
  rrdcached:
    image: crazymax/rrdcached
    container_name: librenms_rrdcached
    ports:
      - target: 42217
        published: 42217
        protocol: tcp
    volumes:
      - "./librenms/rrd:/data/db"
      - "./rrd-journal:/data/journal"
    environment:
      - "TZ=${TZ}"
      - "PUID=${PUID}"
      - "PGID=${PGID}"
      - "LOG_LEVEL=LOG_INFO"
      - "WRITE_TIMEOUT=1800"
      - "WRITE_JITTER=1800"
      - "WRITE_THREADS=4"
      - "FLUSH_DEAD_DATA_INTERVAL=3600"
    restart: always
  librenms:
    image: librenms/librenms:latest
    container_name: librenms
    domainname: 
    hostname: librenms
    ports:
      - target: 8000
        published: 8000
        protocol: tcp
    depends_on:
      - memcached
      - rrdcached
    volumes:
      - "./librenms:/data"
    environment:
      - "TZ=${TZ}"
      - "LIBRENMS_DISTRIBUTED_POLLER_ENABLE=true"
      - "LIBRENMS_DISTRIBUTED_POLLER_NAME=ITSS"
      - "LIBRENMS_DISTRIBUTED_POLLER_GROUP=1"
      - "LIBRENMS_DISTRIBUTED_POLLER_MEMCACHED_HOST="
      - "LIBRENMS_DISTRIBUTED_POLLER_MEMCACHED_PORT=11211"
      - "RRDCACHED_HOST="
      - "RRDCACHED_PORT=42217"
      - "PUID=${PUID}"
      - "PGID=${PGID}"
      - "DB_HOST="
      - "DB_NAME=${MYSQL_DATABASE}"
      - "DB_USER=${MYSQL_USER}"
      - "DB_PASSWORD=${MYSQL_PASSWORD}"
      - "DB_TIMEOUT=60"
    restart: always
  portainer:
    container_name: portainer
    image: portainer/portainer:latest
    restart: unless-stopped
    command: -H unix:///var/run/docker.sock
    ports:
      - "9000:9000"
    volumes:
      - "./portainer:/data"
      - /var/run/docker.sock:/var/run/docker.sock:ro
    environment:
      - "TZ=${TZ}"
      - "PUID=${PUID}"
      - "PGID=${PGID}"
  dispatcher:
    image: librenms/librenms:latest
    container_name: librenms_dispatcher
    domainname: 
    hostname: librenms
    depends_on:
      - librenms
    volumes:
      - "./librenms:/data"
    environment:
      - "TZ=${TZ}"
      - "PUID=${PUID}"
      - "PGID=${PGID}"
      - "DB_HOST=1"
      - "DB_NAME=${MYSQL_DATABASE}"
      - "DB_USER=${MYSQL_USER}"
      - "DB_PASSWORD=${MYSQL_PASSWORD}"
      - "REDIS_HOST="
      - "REDIS_PORT=6379"
      - "REDIS_DB=0"
      - "SIDECAR_DISPATCHER=1"
    restart: always
  dozzle:/
    image: amir20/dozzle:latest
    container_name: dozzle
    restart: unless-stopped
    ports:
      - "9002:8080"
    environment:
      DOZZLE_LEVEL: info
      DOZZLE_TAILSIZE: 300
      DOZZLE_FILTER: "status=running"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
  syslog-ng:
    image: librenms/librenms:latest
    container_name: librenms_syslog
    domainname: 
    hostname: librenms
    depends_on:
      - librenms
    ports:
      - target: 514
        published: 514
        protocol: tcp
      - target: 514
        published: 514
        protocol: udp
    volumes:
      - "./librenms:/data"
    environment:
      - "TZ=${TZ}"
      - "PUID=${PUID}"
      - "PGID=${PGID}"
      - "DB_HOST=172.29.1.31"
      - "DB_NAME=${MYSQL_DATABASE}"
      - "DB_USER=${MYSQL_USER}"
      - "DB_PASSWORD=${MYSQL_PASSWORD}"
      - "DB_TIMEOUT=60"
      - "SIDECAR_SYSLOGNG=1"
    restart: always
```

### Docker info

```
> Output of command `docker info`
Client:
 Debug Mode: false

Server:
 Containers: 9
  Running: 7
  Paused: 0
  Stopped: 2
 Images: 6
 Server Version: 19.03.12
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 4.15.0-108-generic
 Operating System: Ubuntu 18.04.4 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 1
 Total Memory: 7.769GiB
 Name: itssdocker
 ID: URJF:IUWU:TJLX:ABGP:4FZC:ZHZN:CWIK:OLDC:BZ25:EBPT:QTGN:F5VW
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No swap limit support

```

### Logs
Librenms_Dispatch Container Logs:
```
> Container logs (set LOG_LEVEL to debug if applicable)  

today at 2:41 PM [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
today at 2:41 PM [s6-init] ensuring user provided files have correct perms...exited 0.
today at 2:41 PM [fix-attrs.d] applying ownership & permissions fixes...
today at 2:41 PM [fix-attrs.d] done.
today at 2:41 PM [cont-init.d] executing container initialization scripts...
today at 2:41 PM [cont-init.d] 00-fix-logs.sh: executing...
today at 2:41 PM [cont-init.d] 00-fix-logs.sh: exited 0.
today at 2:41 PM [cont-init.d] 01-fix-uidgid.sh: executing...
today at 2:41 PM [cont-init.d] 01-fix-uidgid.sh: exited 0.
today at 2:41 PM [cont-init.d] 02-fix-perms.sh: executing...
today at 2:41 PM Fixing perms...
today at 2:41 PM [cont-init.d] 02-fix-perms.sh: exited 0.
today at 2:41 PM [cont-init.d] 03-config.sh: executing...
today at 2:41 PM Setting timezone to America/Chicago...
today at 2:41 PM Setting PHP-FPM configuration...
today at 2:41 PM Setting PHP INI configuration...
today at 2:41 PM Setting OpCache configuration...
today at 2:41 PM Setting Nginx configuration...
today at 2:41 PM Updating SNMP community...
today at 2:41 PM Initializing LibreNMS files / folders...
today at 2:41 PM Setting LibreNMS configuration...
today at 2:41 PM Fixing perms...
today at 2:41 PM Checking additional Monitoring plugins...
today at 2:41 PM [cont-init.d] 03-config.sh: exited 0.
today at 2:41 PM [cont-init.d] 04-svc-main.sh: executing...
today at 2:41 PM [cont-init.d] 04-svc-main.sh: exited 0.
today at 2:41 PM [cont-init.d] 05-svc-dispatcher.sh: executing...
today at 2:41 PM >>
today at 2:41 PM >> Sidecar dispatcher container detected
today at 2:41 PM >>
today at 2:41 PM Waiting 60s for database to be ready...
today at 2:41 PM Database ready!
today at 2:41 PM [cont-init.d] 05-svc-dispatcher.sh: exited 0.
today at 2:41 PM [cont-init.d] 06-svc-cron.sh: executing...
today at 2:41 PM [cont-init.d] 06-svc-cron.sh: exited 0.
today at 2:41 PM [cont-init.d] 07-svc-syslogng.sh: executing...
today at 2:41 PM [cont-init.d] 07-svc-syslogng.sh: exited 0.
today at 2:41 PM [cont-init.d] 08-svc-weathermap.sh: executing...
today at 2:41 PM [cont-init.d] 08-svc-weathermap.sh: exited 0.
today at 2:41 PM [cont-init.d] done.
today at 2:41 PM [services.d] starting services
today at 2:41 PM [services.d] done.
today at 2:41 PM MainThread(INFO):Configuring LibreNMS service
today at 2:41 PM MainThread(INFO):Attempting to load .env from '/opt/librenms/.env'
today at 2:41 PM librenms(INFO):Attaching signal handlers on thread librenms
today at 2:41 PM librenms(INFO):Using pure python SQL client
today at 2:41 PM librenms(INFO):Created redis lock manager with socket_timeout of 60s
today at 2:41 PM librenms(INFO):Watchdog is disabled.
today at 2:41 PM librenms(INFO):Entering main LibreNMS service loop on 325/librenms...
today at 2:41 PM librenms(INFO):Groups: [0]
today at 2:41 PM librenms(INFO):Poller QueueManager created: 24 workers, 300s frequency
today at 2:41 PM Poller_0-1(INFO):Creating queue poller:0
today at 2:41 PM librenms(INFO):Groups: [0]
today at 2:41 PM librenms(INFO):Discovery QueueManager created: 16 workers, 21600s frequency
today at 2:41 PM Discovery_0-1(INFO):Creating queue discovery:0
today at 2:41 PM librenms(INFO):Using pure python SQL client
today at 2:41 PM Poller_0-1(INFO):Created redis queue with socket_timeout of 60s
today at 2:41 PM Discovery_0-1(INFO):Created redis queue with socket_timeout of 60s
today at 2:41 PM librenms(INFO):Groups: [0]
today at 2:41 PM librenms(INFO):Alerting QueueManager created: 1 workers, 60s frequency
today at 2:41 PM Alerting(INFO):Creating queue alerting:0
today at 2:41 PM librenms(INFO):Using pure python SQL client
today at 2:41 PM Alerting(INFO):Created redis queue with socket_timeout of 60s
today at 2:41 PM librenms(INFO):Groups: [0]
today at 2:41 PM librenms(INFO):Services QueueManager created: 8 workers, 300s frequency
today at 2:41 PM Services_0-1(INFO):Creating queue services:0
today at 2:41 PM librenms(INFO):Using pure python SQL client
today at 2:41 PM Services_0-1(INFO):Created redis queue with socket_timeout of 60s
today at 2:41 PM librenms(INFO):Groups: [0]
today at 2:41 PM librenms(INFO):Billing QueueManager created: 2 workers, 300s frequency
today at 2:41 PM Billing(INFO):Creating queue billing:0
today at 2:41 PM librenms(INFO):LibreNMS Service: librenms-b9be22aa-bb09-11ea-986f-0242ac130007 started!
today at 2:41 PM librenms(INFO):Poller group 0 (default). Using Python 3.8.2 and redis locks and queues
```

**Comment by crazy-max** [comment]
@BraydenBrekke `- "RRDCACHED_HOST="` must be defined and **not empty**.

**Comment by BraydenBrekke** [comment]
yes, we have all host values filled out on our docker-compose file. I simply removed it since we are referencing another rrd cache host that is accessed through a public IP.

**Comment by murrant** [comment]
I suggest you use the WEBUI to configure.

**Comment by BraydenBrekke** [comment]
How would I go about moving individual pollers into poller groups through the GUI? From my understanding, I can only change the value of the default poller in the GUI. 

Even if I am able to move a poller into a group through the GUI though, It still makes it inconvenient when deploying multiple pollers. I would like to just use the docker compose to choose what polling group a poller is in.

**Comment by bfarney** [comment]
I'm confused as well. We are using the Web UI to add poller groups. But I'm very lost on how to assign pollers to said groups via the Web UI. I agree with Brayden it would be great to have it via the docker compose but I'd also be happy to use the Web UI if I knew how.

What I'm finding in the documentation: 
https://docs.librenms.org/Extensions/Distributed-Poller/
https://docs.librenms.org/Extensions/Dispatcher-Service/

Both talk about setting the distributed poller group in the config. 

$config['distributed_poller_group']              = 0;               # Which group to poll

Thanks so much for your time! We really appreciate it!!

**Comment by murrant** [comment]
Left side selects the poller, right side sets the settings, such as the groups.
![image](https://user-images.githubusercontent.com/39462/86265144-0fe80900-bb89-11ea-926f-8644fbba5cad.png)

Make sure you have the groups defined on the groups tab.

**Comment by bfarney** [comment]
Oh snap. I don't even have that tab! :)

![image](https://user-images.githubusercontent.com/30095564/86265739-cb10a200-bb89-11ea-8c3e-43a0fcdb1d15.png)

Here is my about page:

![image](https://user-images.githubusercontent.com/30095564/86265852-f5faf600-bb89-11ea-90e3-5647e4a3e653.png)

Does that help?

Thanks so much!!

**Comment by murrant** [comment]
oh, I guess that is in the 1.65 release, due out any day.

**Comment by crazy-max** [comment]
I think this issue is linked to #93

**Comment by bfarney** [comment]
@crazy-max I can confirm that we also see the symptoms presented in #93 in ours as well. We only get one poller listed in the Poller Cluster Health page. I think I was assuming that the poller needed to poll it its own group first, then it would show up in the poller list. But it seems from the info in #93 that the NODE_ID is the root issue.

Let me know if there is any info you might want from us that can help confirm for you!! 

And thanks again for your help and time.

**Comment by crazy-max** [comment]
@bfarney I'm working on it, keep you in touch.

**Comment by crazy-max** [comment]
The `librenms/librenms:edge` should fix your issue:

![image](https://user-images.githubusercontent.com/1951866/86487526-dd3b3d80-bd4d-11ea-802f-2e0aab9c4fb0.png)

The env var [`LIBRENMS_SERVICE_NODE_ID`](https://github.com/librenms/docker/blob/master/doc/docker/environment-variables.md#dispatcher-service) has been introduced to define a node ID for your dispatcher(s).

**Comment by crazy-max** [comment]
Should be fixed through 1.65. See [CHANGELOG](https://github.com/librenms/docker/blob/master/CHANGELOG.md#165-rc1-20200703) for more info.
